tests/dragonfly/connection_test.py

```diff
@@ -288,47 +288,40 @@ def generate(max):
 """
 
 
-async def test_pubsub_subcommand_for_numsub(async_client):
-    subs1 = [async_client.pubsub() for i in range(5)]
-    for s in subs1:
-        await s.subscribe("channel_name1")
-    result = await async_client.pubsub_numsub("channel_name1")
-    assert result[0][0] == "channel_name1" and result[0][1] == 5
-
-    for s in subs1:
-        await s.unsubscribe("channel_name1")
-    result = await async_client.pubsub_numsub("channel_name1")
-
-    retry = 5
-    for i in range(0, retry):
-        result = await async_client.pubsub_numsub("channel_name1")
-        if result[0][0] == "channel_name1" and result[0][1] == 0:
-            break
+async def test_pubsub_subcommand_for_numsub(async_client: aioredis.Redis):
+    async def resub(s: "aioredis.PubSub", sub: bool, chan: str):
+        if sub:
+            await s.subscribe(chan)
         else:
-            time.sleep(1)
+            await s.unsubscribe(chan)
+        # Wait for PUSH message to be parsed to make sure upadte was performed
+        await s.get_message(timeout=0.1)
```

Comment:
nit: consider returning the message and then asserting in the test

Reply:
I don't use it anywhere, unsubscribe is discarded internally